src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1120,19 +1111,19 @@ init_fd_list(void)
 {
 	/* daos_init() is not called yet, so we can not use D_ERROR() etc. */
 	if (pthread_mutex_init(&lock_fd, NULL) != 0) {
-		D_ERROR("mutex create_new_lock lock_fd init failed\n");
+		D_ERROR("mutex create_new_lock lock_fd init failed: %s\n", strerror(errno));
 		exit(1);
 	}
```

Comment:
Here you can call `rc = D_MUTEX_INIT(&lock_fd, NULL);`, it will log on error for you and you can handle the rc how you wish.

In general when there are errors in the core like this it would be better to continue but return EIO for all intercepted access which cannot be intercepted, exiting the user application directly isn't a good idea normally.